src/System.IO.Pipelines/ref/System.IO.Pipelines.cs

```diff
@@ -62,33 +46,32 @@ public abstract partial class PipeReader
         public abstract void OnWriterCompleted(System.Action<System.Exception, object> callback, object state);
-        public abstract System.IO.Pipelines.PipeAwaiter<System.IO.Pipelines.ReadResult> ReadAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
+        public abstract System.Threading.Tasks.ValueTask<System.IO.Pipelines.ReadResult> ReadAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
         public abstract bool TryRead(out System.IO.Pipelines.ReadResult result);
     }
     public abstract partial class PipeScheduler
     {
         protected PipeScheduler() { }
         public static System.IO.Pipelines.PipeScheduler Inline { get { throw null; } }
         public static System.IO.Pipelines.PipeScheduler ThreadPool { get { throw null; } }
-        public abstract void Schedule(System.Action action);
-        public abstract void Schedule(System.Action<object> action, object state);
+        public abstract void Schedule<TState>(System.Action<TState> action, TState state);
     }
     public abstract partial class PipeWriter : System.Buffers.IBufferWriter<byte>
     {
         protected PipeWriter() { }
         public abstract void Advance(int bytes);
         public abstract void CancelPendingFlush();
         public abstract void Complete(System.Exception exception = null);
-        public abstract System.IO.Pipelines.PipeAwaiter<System.IO.Pipelines.FlushResult> FlushAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
+        public abstract System.Threading.Tasks.ValueTask<System.IO.Pipelines.FlushResult> FlushAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
         public abstract System.Memory<byte> GetMemory(int minimumLength = 0);
         public abstract System.Span<byte> GetSpan(int minimumLength = 0);
         public abstract void OnReaderCompleted(System.Action<System.Exception, object> callback, object state);
-        public virtual System.IO.Pipelines.PipeAwaiter<System.IO.Pipelines.FlushResult> WriteAsync(System.ReadOnlyMemory<byte> source, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual System.Threading.Tasks.ValueTask<System.IO.Pipelines.FlushResult> WriteAsync(System.ReadOnlyMemory<byte> source, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
     public partial struct ReadResult
     {
         private object _dummy;
-        public ReadResult(System.Buffers.ReadOnlySequence<byte> buffer, bool isCanceled, bool isCompleted) { throw null; }
+        public ReadResult(in System.Buffers.ReadOnlySequence<byte> buffer, bool isCanceled, bool isCompleted) { throw null; }
```

Comment:
Does this make a measurable performance impact?  It looks like the struct is two objects and two ints.

Reply:
Haven’t measured this specifically. We were passing around the buffer with in in other places (in System.Memory itself).